### ISSUE TITLE
Reproducing results

### ISSUE THREAD

**Issue by HaoranTang** [canqin001/PointDAN#15]
Hi,

This work is really interesting and insightful! I want to reproduce the experiment and have the following questions:

1. How can we manage the ablation study G, L, A, P as shown in Table 2 and 3? Do we need to enable/disable them in forward function of Net_MDA? I just started training it hours ago without any modification, so I wonder which ablation study should I compare with when finishing training.

2. Do you have any recommended visualization tools for point cloud? I also want to visualize the experiment, too.

Looking forward to your reply, I am a beginner in domain adaptation and there are so many excellent works for me to learn!

**Comment by canqin001** [comment]
Hi, thanks for your interest of our work. For Q1, you could comment the different losses in the training script and see the difference. For Q2, please refer to the 'visualize_sample.py'.

**Comment by HaoranTang** [comment]
Hi, 

Thank you for the update! I can see besides global feature alignment G, there is also local alignment L and its corresponding loss in train.py, but I did not find where represent the SA node attention A and self-training S and not sure how can I modify them to see the difference...I do see SA node part in Model.py, where you pass many Falses to the forward function in Net_MDA but it contains many conditions that run when True. Can I improve the performance by changing those Falses to Trues? 

Besides, I wonder how can we open and see the .DESKTOP files in log folder...want to check history and compare.

I apologize if my questions are overwhelming and somehow naive, and I really appreciate that you are willing to teach me...

**Comment by canqin001** [comment]
If you remove the local alignment, the SA node attention would be useless. The attention layers are defined in the line 105-106 in "Model.py". We do not provide the self-training in this code. It is very easy to implement the self-training in which you just need to select the the top 10% target samples with the highest softmax score. For example, in each batch, rank the samples according to their softmax score. Then select the top 10% one and regard the predicted results as the ground truth. Then, finetune the model by this pseudo labels. BTW, I am also not sure how to open the " .DESKTOP" file.

**Comment by HaoranTang** [comment]
Ok, thank you for this information! I just wonder how we use the parameters passed to forward function (adaptation=False, node_vis=False, mid_feat=False, node_adaptation_s=False, node_adaptation_t=False), because I see they only work when true in the function, but you gave them falses...specially the adaptation and node_adaptation_s/t. Do I need to modify them if I only want to reproduce the best results?

**Comment by canqin001** [comment]
I am a little bit confused which parts of code you are talking about. I give node_adaptation_s/t true in the line 190-191 of "train.py".

**Comment by HaoranTang** [comment]
Yes I saw it last night...sorry I forgot to update, now I understand your code details. Thank you again for explaining to me, great work!

**Comment by canqin001** [comment]
Perfect!
